### ISSUE TITLE
Strange behavior of CLI flags

### ISSUE THREAD

**Issue by jetexe** [urfave/cli#1948]
## Checklist

* [x] Are you running the latest v3 release? The list of releases is [here](https://github.com/urfave/cli/releases).
* [x] Did you check the manual for your release? The v3 manual is [here](https://github.com/urfave/cli/blob/main/docs/v3/index.md).
* [x] Did you perform a search about this feature? Here's the [GitHub guide](https://help.github.com/en/github/managing-your-work-on-github/using-search-to-filter-issues-and-pull-requests) about searching.

## Problem

Currently if you mark your flag as required and try to call a command with `--help` flag you will get an error:

```sh
Required flag "<name>" not set
```

Same for the validation: If you want to print help you will get the validation error. 

## Proposal

I'm suggesting we don't run a validation and required check during help printing

**Comment by BlackHole1** [comment]
I'm not sure if this is very difficult to achieve, but I would like to know if we can provide a switch in version v3. Something like:

```go
&cli.StringFlag{
	Name:               "name",
	Required:           true,
	HelpExcludeRequire: false // Default: true
},
```

or

```go
Name: "config",
HelpExcludeRequireFlags: true
Flags: []cli.Flag{
	&cli.StringFlag{
		Name:        "name",
		Required:    true,
	},
},

```

**Comment by tarampampam** [comment]
Unfortunately, I have the same problem as described 😞

**Comment by jetexe** [comment]
> provide a switch

The problem is in the prepare process. `cli` parses flags and command first, and only after that makes a decision to print the help. I wonder if we can ignore validation and required flags errors in case of help printing

**Comment by dearchap** [comment]
Should be fixed by #1987 . @jetexe please test with latest v3 and let me know,
